internal/slicer/report.go

```diff
@@ -0,0 +1,47 @@
+package slicer
+
+import (
+	"io/fs"
+
+	"github.com/canonical/chisel/internal/fsutil"
+	"github.com/canonical/chisel/internal/setup"
+)
+
+type ReportEntry struct {
+	Path   string
+	Mode   fs.FileMode
+	Hash   string
+	Size   int
+	Slices []*setup.Slice
+	Link   string
+}
+
+// Report holds the information about files and directories created when slicing
+// packages.
+type Report struct {
+	Root string
+	// map indexed by path.
+	Entries map[string]ReportEntry
+}
+
+func NewReport(root string) *Report {
```

Comment:
nit: same as above. Let's document what `NewReport` does.

Reply:
I think it is pretty self explanatory :P, right? AFAIK the approach we take with documentation is to document only items that need it because of complex naming or caveats / assumptions.

Reply:
```
// NewReport returns an empty report for content that will be based at the provided root path.
```
I agree with your principle, to be clear, but some things are not obvious here and can easily be described.